app/course/[course_id]/manage/course/emails/previewAndSend.tsx

```diff
@@ -0,0 +1,317 @@
+          .filter((user) => {
+            return user.role === "student";
+          })
+          .map((user) => {
+            return user.users.user_id;
+          }) ?? [];
+      return studentIdsForClass?.filter((id) => {
+        return !submittedIds.includes(id);
+      });
+    }
+    return [];
+  };
+
+  const getTagIds = (audience: TagEmailInfo) => {
+    const chosenTag = audience.tag;
+    const profile_ids = tags
+      .filter((tag) => {
+        return tag.color == chosenTag.color && tag.name == chosenTag.name && tag.visible === chosenTag.visible;
+      })
+      .map((tag) => {
+        return tag.profile_id;
+      });
+    const ids =
+      userRolesData?.data
+        .filter((user) => {
+          return profile_ids.includes(user.private_profile_id);
+        })
+        .map((user) => {
+          return user.users.user_id;
+        }) ?? [];
+    return ids;
+  };
+
+  const getGeneralIds = (audience: GeneralEmailInfo) => {
+    const roles: string[] = [];
+    if (audience.includeGraders) {
+      roles.push("grader");
+    }
+    if (audience.includeInstructors) {
+      roles.push("instructor");
+    }
+    if (audience.includeStudents) {
+      roles.push("student");
+    }
+    return (
+      userRolesData?.data
+        .filter((userRole) => {
+          return roles.includes(userRole.role);
+        })
+        .map((userRole) => {
+          return userRole.users.user_id;
+        }) ?? []
+    );
+  };
+
+  return (
+    <Box width={{ base: "100%" }}>
+      <Heading size="lg" mt="5" mb="5">
+        Preview and send
+      </Heading>
+      {emailsToCreate.length > 0 ? (
+        <Box spaceY="4">
+          {emailsToCreate.map((email, key) => {
+            return (
+              <Card.Root key={key} padding="2" mt="5" size="sm">
+                <Flex justifyContent={"space-between"}>
+                  <Card.Title>
+                    <Flex alignItems="center">
+                      Subject:
+                      <Editable.Root
+                        value={email.subject}
+                        onValueChange={(e) => {
+                          {
+                            if (email.id) {
+                              updateEmailField(email.id, "subject", e.value);
+                            }
+                          }
+                        }}
+                      >
+                        <Editable.Preview />
+                        <Editable.Input />
+                      </Editable.Root>
+                    </Flex>
+                  </Card.Title>
+                  <Text
+                    onClick={() => {
+                      if (email.id) {
+                        removeEmail(email.id);
+                      }
+                    }}
+                  >
+                    <IoMdClose />
+                  </Text>
+                </Flex>
+                <Flex flexDir={"column"} fontSize="sm">
+                  <Text>To: {emailToAudienceText(email)}</Text>
```

Comment:
For tags, this throws browser error:

```
hook.js:608 In HTML, <div> cannot be a descendant of <p>.
This will cause a hydration error.
```

I think that it would be fine for this to be a `<Box>`, which would solve this error.